src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp

```diff
@@ -109,22 +69,22 @@ struct convolution_onednn : typed_primitive_onednn_impl<convolution, dnnl::convo
 
         if (arg.activations_zero_points_term()) {
             auto& a_zp = arg.activations_zero_points();
+            auto a_zp_data_type = a_zp.get_output_layout().data_type;
 
-            memory::ptr s32_mem;
-            if (a_zp.get_output_layout().data_type == data_types::i8) {
-                onednn::make_per_tensor_if_possible<data_type_to_type<data_types::i8>::type>(a_zp.as<data>());
-                s32_mem = onednn::convert_zp_data_to_s32<data_type_to_type<data_types::i8>::type>(a_zp.as<data>().get_attached_memory_ptr());
-            } else if (a_zp.get_output_layout().data_type == data_types::u8) {
-                onednn::make_per_tensor_if_possible<data_type_to_type<data_types::u8>::type>(a_zp.as<data>());
-                s32_mem = onednn::convert_zp_data_to_s32<data_type_to_type<data_types::u8>::type>(a_zp.as<data>().get_attached_memory_ptr());
-            } else {
+            if (!data_type_traits::is_i8_u8(a_zp_data_type)) {
                 throw std::runtime_error("Unsupported data type for activations zero points for oneDNN convolution");
             }
-            a_zp.as<data>().attach_memory(s32_mem, false);
 
-            int mask = a_zp.get_output_layout().count() > 1 ? 2 : 0;
+            bool is_per_tensor = true;
+            std::vector<int32_t> zp_values;
+            if (a_zp_data_type == data_types::i8) {
+                is_per_tensor = onednn::is_per_tensor<data_type_to_type<data_types::i8>::type>(a_zp.as<data>(), zp_values);
+            } else { // if (a_zp_data_type == data_types::u8)
+                is_per_tensor = onednn::is_per_tensor<data_type_to_type<data_types::u8>::type>(a_zp.as<data>(), zp_values);
+            }
 
-            attrs->set_zero_points(DNNL_ARG_SRC, mask, {DNNL_RUNTIME_S32_VAL});
+            int mask = is_per_tensor ? 0 : 2;
+            attrs->set_zero_points(DNNL_ARG_SRC, mask, zp_values);
```

Comment:
I\`m the author of this proposal, so I\`ll comment.\
`set_zero_points` is only preferable when `is_per_tensor` is true (i.e. when the ZP is scalar). In other cases, please still use the GPU buffer, for it saves oneDNN time it might spend creating the buffer itself and copying the data there.

Reply:
I got it.